File: algorithms/522. Longest Uncommon Subsequence II/solution.cc
class Solution {
public:
    int findLUSlength(vector<string>& strs) {
        int n = strs.size();
        int lus = -1;
        int i, j;
        for ( i = 0; i < n; i++ ) {
            for ( j = 0; j < n; j++ ) {
                if ( i != j && is_subseq(strs[i], strs[j]) ) {
                    break;
                }
            }
            if ( j == n ) {
                lus = max(lus, (int)strs[i].size());
            }
        }
        return lus;
    }

    bool is_subseq(string s1, string s2) {
        int i = 0;
        for ( int j = 0; j < s2.size(); j++ ) {
            if ( i < s1.size() && s1[i] == s2[j] ) {
                i++;
            }
        }
        return i == s1.size();
    }
};

File: algorithms/522. Longest Uncommon Subsequence II/solution.py
class Solution(object):
    def findLUSlength(self, strs):
        """
        :type strs: List[str]
        :rtype: int
        """
        def is_subseq(s1, s2):
            '''s1 is subsequence of s2'''
            i = 0
            for c in s2:
                if i < len(s1) and c == s1[i]:
                    i += 1
            return i == len(s1)

        strs = sorted(strs, key=len, reverse=1)
        for i in xrange(len(strs)):
            for j in xrange(len(strs)):
                if i != j and is_subseq(strs[i], strs[j]):
                    break
            else:
                return len(strs[i])
        return -1
